File: Data Structures & Algorithms/house-robber-ii/submission-1.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 1) return nums[0];
        return Math.max(houserob(nums, 0), houserob(nums, 1));
    }

    int houserob(int[] nums, int start) {
        int rob1 = 0, rob2 = 0;
        for (int i = start; i < nums.length + start - 1; i++) {
            int temp = Math.max(nums[i] + rob1, rob2);
            rob1 = rob2;
            rob2 = temp;
        }
        return rob2;
    }
}

File: Data Structures & Algorithms/remove-element/submission-1.go
func removeElement(nums []int, val int) int {
    n := len(nums)
	for n > 0 && nums[n-1] == val {
		n--
	}

	for i := 0; i < n;  {
		if nums[i] == val {
			n--
			nums[i] = nums[n]
		} else {
			i++
		}
	}
	return n
}
